README.MD

```diff
@@ -101,31 +105,132 @@ replacing the `specimin.tool.path` placeholder.
       ```
 
 **Command Arguments:**
+
 - **--root:** Root directory of the target project.
     - Example: `/root/path/to/your/targeted/project`
 - **--targetFile:** Source file to search for target methods.
     - Example: `path/to/targetFile/Example.java`
-- **--targetMethod:** Target method to preserve. Format: 
-`class.fully.qualified.Name#methodName(Param1Type, Param2Type, ...)`.
+- **--targetMethod:** Target method to preserve. Format:
+  `class.fully.qualified.Name#methodName(Param1Type, Param2Type, ...)`.
     - Example: `com.example.Example#foo()`
 
 ### Logging
+
 > 🟡 **Note:** All logs are written to the `logs` directory in the project root directory.
 
 <br/>
 
 ---
 
+## Automation Process
+
+ASHE includes an automation layer that streamlines the application of its hardening mechanisms on Java files
+and repositories. This is accomplished through two primary components:
+
+### AsheAutomation Class
+
+The AsheAutomation class in edu.njit.jerse.automation automates the application of ASHE's minimization and
+error correction features on Java files within a specified directory.
+
+#### Key functionalities include:
+
+- Processing All Java Files: Iterates over all Java files in a given Java source directory, applying ASHE's
+  mechanisms to each file. The Java source directory is expected to match Maven/Gradle's standards,
+  where the source code is located in `src/main/java`.
+- Method-Level Precision: Targets public methods in public classes within the Java files for minimization
+  and error correction.
+
+### RepositoryAutomationEngine Class
+
+The RepositoryAutomationEngine class in edu.njit.jerse.automation handles the automation of repository
+cloning and Java file processing.
+
+#### Its main features are:
+
+- Repository Management: Clones or fetches repositories listed in a provided CSV file.
+- Bulk Processing: Applies AsheAutomation to every Java file within the cloned repositories.
+- Customizable Repository Data: Accepts a CSV file with repository URLs and branch information for
+  targeted processing.
+
+<br/>
+
+---
+
+## Automation Usage
+
+/absolute/path/to/project/src/main/java/com/example/foo
+/absolute/path/to/project
+
+### Compiling ASHE
+
+1. **Navigate to project directory:**
+    - Open your terminal and navigate to the project root directory using the `cd` command.
+2. **Compile ASHE:**
+    - Execute the following command:
+      ```bash
+      javac -cp ".:../resources:../../../libs/*" edu/njit/jerse/*.java
```

Comment:
Shouldn't you run a Gradle command instead of compiling the java files by hand?

Reply:
This may require some gradle modifcation. Since this portion is unrelated to automate-ashe, would it make more sense to do so in a separate PR?

Reply:
Disregard last comment - I noticed this was within the newly added automation process. I thought this was from the original portion of the readme.